### DISCUSSION TITLE
use custom Mitsuba path - what entry?

### DISCUSSION THREAD

**Discussion by orarugogerincufelpattano** [mitsuba-renderer/mitsuba-blender#90] [General]
Hi!
I'm on a company network and therefore the installation of Mitsuba is a big challenge.
I installed Python 312 for my user only (as I have no Admin rights).
It was installed in to:  
`C:\Users\xxxxxxx\AppData\Local\Programs\Python\Python312`
Then I downloaded wheel versions of Mitsuba and Dr.jit and installed with the local option:
`pip install --no-index --find-links=file:./ drjit-0.4.4-cp312-cp312-win_amd64.whl mitsuba-3.5.0-cp312-cp312-win_amd64.whl`
They got installed in to:
`C:\Users\xxxxxxx\AppData\Local\Programs\Python\Python312\Scripts   (mitsuba.exe)`
and:
`C:\Users\xxxxxxx\AppData\Local\Programs\Python\Python312\Lib\site-packages    ( drjit; drjit-0.4.4.dist-info;  mitsuba;  mitsuba-3.5.0.dist-info)`
(When I try to use the PIP Installation inside the add-on, it brakes down with time out after a while. 
Looks like the sites are blocked on the network. That's why I used the local installation method.)

I can start mitsuba from anywhere so it seems to be in the PATH but I can't figure out what entry I have to make in the Mitsuba Blender Add-On. Whatever I write in to the "Use custom Mitsuba path" field, it doesn't work.
I modified the __init__.py from the add-on and changed **DEPS_MITSUBA_VERSION** = to '**3.5.0**'
But this also not helped.
Can you give me a hint to how to make the plugin work on this awkward installation?
Thanks.

Sorry, I forgot to write: I'm using Blender 3.6.5 LTS version.

**Comment by orarugogerincufelpattano** [comment]
Can someone comment on what packages pip is looking for if I try to install from inside the add-on?
I modified the __init__.py and changed the pip command to local:
`result = subprocess.run([sys.executable, '-m', 'pip', 'install', f'mitsuba=={DEPS_MITSUBA_VERSION}',  '--user', '--force-reinstall', '--no-index', '--find-links', 'file:///local_lib/Python_ZIPs'], capture_output=False)`
I put drjit-0.4.4-cp312-cp312-win_amd64.whl mitsuba-3.5.0-cp312-cp312-win_amd64.whl in the specified location but if I try to install, pip tells me that "No matching distribution found for mitsuba==3.5.0"
In the original add-on the version number for Mitsuba was 3.4.1 but Mitsuba has now version 3.5.0.
Is that a problem?
Which DrJit package version belongs to Mitsuba 3.4.1?

**Comment by orarugogerincufelpattano** [comment]
Oh boy... what a mess...
I can install python and pip for Windows as often as I want but the Add-On doesn't care about that!
I increased the verbosity level of the pip command in the init.py of the Add-on and so I could see that the Add-On is using Blenders built in Python:
`Using pip 23.0.1 from C:\local_lib\_BLENDER\blender-3.6.8-windows-x64\3.6\python\lib\site-packages\pip (python 3.10)`
That was the reason why the local distribution never matched. 
I could have spared two days if there would be a hint in the documentation to this!

**Comment by bathal1** [comment]
Yes, blender comes bundled with its own version of python. The add-on will run `python -m pip install mitsuba==3.4.1` using this python executable. I'm not sure if that will solve your permissions issues though.

The other option is to compile Mitsuba from source, and provide the addon with the path to the build/Release directory.

**Reply by orarugogerincufelpattano** [reply]
It's all solved now and the add-on displays no error anymore, thanks. 👍
Tomorrow, I will make a test export.

I was not aware that the Add-On uses the Blender built in python version which is an older version 3.10 instead of 3.12.
I thought it uses python from the system where I installed 3.12 with the corresponding version of Mitsuba.
I changed the Add-On line with the pip command to install a local wheel of Mitsuba but that had 312 as a version. It stopped all the time complaining that the version number doesn't match... Sure, not to the Blender built in python.
I guess there are not many people who have such problems (sitting behind very restrictive company firewalls) but a short mention of this fact in the Add-On doc page would be maybe helpful.

Compiling was impossible as all the prerequisites needed administrator installation.
